Documentation: Created XML Sitemap Post (#10106)
gatsbyjs/gatsby: docs/docs/creating-a-sitemap.md

---
title: Creating a sitemap
---
### What is a Site Map?

An [XML sitemap](https://support.google.com/webmasters/answer/156184?hl=en) lists a website’s important pages, making sure search engines (such as Google) can find and crawl them all, and helping it understand your website structure.

Think of it as a map for your website. It shows what all of the pages are on your website.

### Install

> To generate a XML sitemap, We will use the [gatsby-plugin-sitemap](/packages/gatsby-plugin-sitemap/) Package, To install this package, Run the following line on your command line:

Reviewer: ```suggestion
To generate an XML sitemap, we will use the [`gatsby-plugin-sitemap`](/packages/gatsby-plugin-sitemap/) package. To install this package, run the following line on your command line:
```

>    ```npm install --save gatsby-plugin-sitemap```

Reviewer: You can highlight console commands like so:

~~~md
```bash
npm install --save gatsby-plugin-sitemap
```
~~~

The formatting is also a bit off here, but that will likely be auto-fixed by Prettier.

### How to use [gatsby-plugin-sitemap](/packages/gatsby-plugin-sitemap/)
````javascript

> // In your gatsby-config.js

Reviewer: ```suggestion
```javascript:title=gatsby-config.js
```

siteMetadata: {
  siteUrl: `https://www.example.com`,
},
plugins: [`gatsby-plugin-sitemap`]
````
Above is the minimal configuration required to have it work. By default, the generated sitemap will include all of your site’s pages.

### Additonal Modification

> Additonal Modification steps are available in the [gatsby-plugin-sitemap documentation](https://www.gatsbyjs.org/packages/gatsby-plugin-sitemap)

Reviewer: ```suggestion
Additonal modification steps are available in the [`gatsby-plugin-sitemap` documentation](https://www.gatsbyjs.org/packages/gatsby-plugin-sitemap)
```

title: Creating a sitemap
---

> ### What is a Site Map?

Reviewer 2: Why Site Map vs. sitemap? Which is correct?

I think let's be consistent and always go one word, e.g.

```suggestion
### What is a Sitemap?
```

> An [XML sitemap](https://support.google.com/webmasters/answer/156184?hl=en) lists a website’s important pages, making sure search engines (such as Google) can find and crawl them all, and helping it understand your website structure.

Reviewer 2: ```suggestion
An [XML sitemap](https://support.google.com/webmasters/answer/156184?hl=en) lists a website’s important pages, making sure search engines (such as Google) can find and crawl them all. In effect, a sitemap helps a search engine understand your website structure.
```

Think of it as a map for your website. It shows what all of the pages are on your website.

### Install

> To generate an XML sitemap, you will use the [`gatsby-plugin-sitemap`](/packages/gatsby-plugin-sitemap/) package. To install this package, run the following line on your command line:

Reviewer 2: ```suggestion
To generate an XML sitemap, you will use the [`gatsby-plugin-sitemap`](/packages/gatsby-plugin-sitemap/) package. To install this package, run the following command:
```

`npm install --save gatsby-plugin-sitemap`

> ### How to use [gatsby-plugin-sitemap](/packages/gatsby-plugin-sitemap/)

Reviewer 2: Perhaps transition into this section? e.g.

```md
Once installed, we can now add this plugin to our `gatsby-config.js`, like so:
```

Author: Good idea! I will reflect something like that on the next commit

> ````javascript

Reviewer 2: This code block is a little messed up. It's using a JavaScript fenced code block, but the contents inside are Markdown. I think you just need to remove the second level of backticks _and_ change to use 3 backticks instead of 4.

Author: Hi,
Could you clarify that step? I might be silly but I don't know what to do for that step 🤦‍♂️

Sorry!

Reviewer 2: Something like this!

````markdown
```javascript:title=gatsby-config.js
module.exports = {
  siteMetadata: {
    siteUrl: `https://www.example.com`,
  },
  plugins: [`gatsby-plugin-sitemap`]
}
````

Author: Oh I gotcha now! Thank you so much!

```javascript:title=gatsby-config.js
siteMetadata: {
  siteUrl: `https://www.example.com`,
},
plugins: [`gatsby-plugin-sitemap`]
````

> Above is the minimal configuration required to have it work. By default, the generated sitemap will include all of your site’s pages.

Reviewer 2: ```suggestion
This is all that's required to get a working sitemap with Gatsby! By default, the generated sitemap will include all of your site’s pages, but of course the plugin exposes options to configure this default functionality.
```

> ### Additonal Modification

Reviewer 2: ```suggestion
### Additional Modification
```

> Additonal modification steps are available in the [`gatsby-plugin-sitemap` documentation](https://www.gatsbyjs.org/packages/gatsby-plugin-sitemap)

Reviewer 2: ```suggestion
Additional modification steps are available in the [`gatsby-plugin-sitemap` documentation](/packages/gatsby-plugin-sitemap)
```

Author: Thanks for picking up to this! I don't know how I lost the ability to spell additional!!
